linc_convert/utils/zarr/generate_pyramid.py

```diff
@@ -0,0 +1,166 @@
+import itertools
+from typing import Literal
+
+import numpy as np
+import zarr
+
+from linc_convert.utils.math import ceildiv
+
+
+def generate_pyramid(
+    omz: zarr.Group,
+    levels: int = -1,
+    ndim: int = 3,
+    max_load: int = 512,
+    mode: Literal["mean", "median"] = "median",
+    no_pyramid_axis: int | None = None,
+) -> list[list[int]]:
+    """
+    Generate the levels of a pyramid in an existing Zarr.
+
+    Parameters
+    ----------
+    omz : zarr.Group
+        Zarr group object
+    levels : int
+        Number of additional levels to generate.
+        By default, stop when all dimensions are smaller than their
+        corresponding chunk size.
+    ndim : int
+        Number of spatial dimensions.
+    max_load : int
+        Maximum number of voxels to load along each dimension.
```

Comment:
Keep it for later, but this should be based on the chunk/stride size, as discussed.